indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java

```diff
@@ -232,9 +240,15 @@ private IndexDataReadResult readIndexMT(IndexWriter w, IndexingContext context)
         }
 
         if (!errors.isEmpty()) {
-            IOException exception = new IOException("Error during load of index");
-            errors.forEach(exception::addSuppressed);
-            throw exception;
+            if (errors.stream().allMatch(ex -> ex instanceof IOException || ex instanceof InterruptedException)) {
+                IOException exception = new IOException("Error during load of index");
+                errors.forEach(exception::addSuppressed);
+                throw exception;
+            } else {
+                RuntimeException exception = new RuntimeException("Error during load of index");
+                errors.forEach(exception::addSuppressed);
+                throw exception;
+            }
         }
```

Comment:
I thought it is probably better to rethrow non-checked exceptions as RT exception (?)